chore: merge dev improvements into master

akloeckner opened the pull request:
@ildar170975, any objections to issue a new "real" release? (After some preliminary releases on `dev`...)

ildar170975 committed:
fix: indicator style for the 1st state (#1105)

arychj committed:
fix: legend indicators respect entity attribute configuration (#1115)

dependabot[bot] committed:
build(deps): bump amannn/action-semantic-pull-request (#1112)

Bumps the patch-and-minor group with 1 update in the / directory: [amannn/action-semantic-pull-request](https://github.com/amannn/action-semantic-pull-request).


Updates `amannn/action-semantic-pull-request` from 5.4.0 to 5.5.3
- [Release notes](https://github.com/amannn/action-semantic-pull-request/releases)
- [Changelog](https://github.com/amannn/action-semantic-pull-request/blob/main/CHANGELOG.md)
- [Commits](https://github.com/amannn/action-semantic-pull-request/compare/v5.4.0...v5.5.3)

---
updated-dependencies:
- dependency-name: amannn/action-semantic-pull-request
  dependency-type: direct:production
  update-type: version-update:semver-minor
  dependency-group: patch-and-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

semantic-release-bot committed:
chore(release): 0.12.2-dev.1 [skip ci]

## [0.12.2-dev.1](https://github.com/kalkih/mini-graph-card/compare/v0.12.1...v0.12.2-dev.1) (2024-07-24)

### Bug Fixes

* indicator style for the 1st state ([#1105](https://github.com/kalkih/mini-graph-card/issues/1105)) ([c0ad333](https://github.com/kalkih/mini-graph-card/commit/c0ad333ee96998bab7179b23b86684ba74851326))
* legend indicators respect entity attribute configuration ([#1115](https://github.com/kalkih/mini-graph-card/issues/1115)) ([5fe01d1](https://github.com/kalkih/mini-graph-card/commit/5fe01d15a791d09646fd26cce5c8d7113e0cd4b7))

akloeckner committed:
use d3-interpolate to interpolate color in main.js

akloeckner committed:
Revert "use d3-interpolate to interpolate color in main.js"

This reverts commit 7e6290b15c2ae043884263ad0de583750b1f5f0e.

I erroneously pushed it. Sorry.

akloeckner committed:
fix: use d3 to interpolate colors (#1118)

semantic-release-bot committed:
chore(release): 0.12.2-dev.2 [skip ci]

## [0.12.2-dev.2](https://github.com/kalkih/mini-graph-card/compare/v0.12.2-dev.1...v0.12.2-dev.2) (2024-08-07)

### Bug Fixes

* use d3 to interpolate colors ([#1118](https://github.com/kalkih/mini-graph-card/issues/1118)) ([b8676f5](https://github.com/kalkih/mini-graph-card/commit/b8676f557107fb5ebda550652a947fa74ab39a9c))

akloeckner committed:
fix: name and icon color respect attribute choice (#1131)

JulienDeveaux committed:
feat: add show_legend_state (#1173)



---------

Co-authored-by: Leonardo Merza <ljmerza@gmail.com>

JulienDeveaux committed:
feat: Add loader (#1180)

semantic-release-bot committed:
chore(release): 0.13.0-dev.1 [skip ci]

# [0.13.0-dev.1](https://github.com/kalkih/mini-graph-card/compare/v0.12.2-dev.2...v0.13.0-dev.1) (2024-12-21)

### Bug Fixes

* name and icon color respect attribute choice ([#1131](https://github.com/kalkih/mini-graph-card/issues/1131)) ([cbfba7a](https://github.com/kalkih/mini-graph-card/commit/cbfba7a6a4a02e944d82d83fca69f61148cbcd34))

### Features

* Add loader ([#1180](https://github.com/kalkih/mini-graph-card/issues/1180)) ([30c5263](https://github.com/kalkih/mini-graph-card/commit/30c5263a30dfd2ebde6b4d1a4969696073c20482))
* add show_legend_state ([#1173](https://github.com/kalkih/mini-graph-card/issues/1173)) ([3b1c827](https://github.com/kalkih/mini-graph-card/commit/3b1c82793a01baf613dfae4ad4bfa981a29b999f))

FAB1150 committed:
fix: more intuitive min_bound_range behavior (#1136)

Now if one bound is defined but a soft bound, and there is no other bound defined, it treats it as a bound.

If both are soft bounds then it doesn't care.

And if the other bound is a hard bound, the soft one is breached.

ildar170975 committed:
fix: remove unused argument from getBoundary (#1193)

ildar170975 committed:
fix: legend unit percent w/o whitespace (#1191)

if UoM = "%", then do not use a whitespace as a delimeter.

ildar170975 committed:
fix: hide graph loading indicator when appropriate (#1197)

ildar170975 committed:
feat: add "tooltip--label" class (#1202)

galo2099 committed:
fix: adapt state color to tooltip properties (#1214)

semantic-release-bot committed:
chore(release): 0.13.0-dev.2 [skip ci]

# [0.13.0-dev.2](https://github.com/kalkih/mini-graph-card/compare/v0.13.0-dev.1...v0.13.0-dev.2) (2025-02-15)

### Bug Fixes

* adapt state color to tooltip properties ([#1214](https://github.com/kalkih/mini-graph-card/issues/1214)) ([1142f25](https://github.com/kalkih/mini-graph-card/commit/1142f259b168ddc936e7060c6cffabc4e89dc8bf))
* hide graph loading indicator when appropriate ([#1197](https://github.com/kalkih/mini-graph-card/issues/1197)) ([d708d6a](https://github.com/kalkih/mini-graph-card/commit/d708d6a22f1531681855a1b852f2f7ee7cfc07ba))
* legend unit percent w/o whitespace ([#1191](https://github.com/kalkih/mini-graph-card/issues/1191)) ([9f5cfd9](https://github.com/kalkih/mini-graph-card/commit/9f5cfd9c86e1d05bbaa14691fbaa77bceb63144a))
* more intuitive min_bound_range behavior ([#1136](https://github.com/kalkih/mini-graph-card/issues/1136)) ([54d9875](https://github.com/kalkih/mini-graph-card/commit/54d987568b850451e0b6009d3d002526bb1f5b13))
* remove unused argument from getBoundary ([#1193](https://github.com/kalkih/mini-graph-card/issues/1193)) ([f5261d9](https://github.com/kalkih/mini-graph-card/commit/f5261d92c37b93f2fd8c7f8da01fe8af4a5a18e8))

### Features

* add "tooltip--label" class ([#1202](https://github.com/kalkih/mini-graph-card/issues/1202)) ([0d3c184](https://github.com/kalkih/mini-graph-card/commit/0d3c184dc19c43dd2c0797f3f7d25b6aaac014e4))

mxr committed:
docs: add some syntax highlighting to README (#1224)

ildar170975 committed:
fix: computeUom() for attribute & empty unit (#1236)

ildar170975 committed:
fix computeName() to fall back to entity ID (#1235)

ildar170975 committed:
fix: padding fill (#1238)

ildar170975 committed:
fix: use a proper config in renderState() (#1217)

ildar170975 committed:
fix: process absent UoM in computeLegend() (#1237)

semantic-release-bot committed:
chore(release): 0.13.0-dev.3 [skip ci]

# [0.13.0-dev.3](https://github.com/kalkih/mini-graph-card/compare/v0.13.0-dev.2...v0.13.0-dev.3) (2025-04-29)

### Bug Fixes

* computeUom() for attribute & empty unit ([#1236](https://github.com/kalkih/mini-graph-card/issues/1236)) ([436b9a9](https://github.com/kalkih/mini-graph-card/commit/436b9a94f3b16f60fa39bcce5163ee07038c7e22))
* padding fill ([#1238](https://github.com/kalkih/mini-graph-card/issues/1238)) ([f930fa6](https://github.com/kalkih/mini-graph-card/commit/f930fa60dfce7b1003b2ebb64c9ce1673b160c60))
* process absent UoM in computeLegend() ([#1237](https://github.com/kalkih/mini-graph-card/issues/1237)) ([5724cfd](https://github.com/kalkih/mini-graph-card/commit/5724cfdd8e73316659d4bd38ee351628f247ae85))
* use a proper config in renderState() ([#1217](https://github.com/kalkih/mini-graph-card/issues/1217)) ([c980460](https://github.com/kalkih/mini-graph-card/commit/c9804600989bd9d3c5868e0312479a26330ac7f2))

ildar170975 commented:
Why not?)))
We will add next changes into next dev release, let already merged things be released “officially”. 

But - I would merge a PR for the loading indicator as well. The old indicator was removed from HA.

akloeckner commented:
Stalling to wait for #1234 (best issue number ever)

ildar170975 committed:
fix: deprecated paper-item-icon-color (#1247)

DJaeger committed:
docs: add "My Home Assistant" HACS Link to README (#1253)

ildar170975 committed:
fix: migrate to ha-spinner  (#1234)

Remove the old spinner and use the new one. The old spinner was discontinued in HA 2025.5.

The loading indicator will not be shown, if your are using a version of HA < 2025.5!

akloeckner commented:
> Can we talk in Community DM?

Sorry, this just showed up, when I had hit the merge button.

ildar170975 commented:
> Sorry, this just showed up

No problem, just thought can assist you.
